### PR TITLE
refactoring of db abstraction

### PR DESCRIPTION
**Pull request by shecken**

### CODE REVIEW

File: classes/Settings/DB.php [hunk 1]
```diff
@@ -18,35 +18,34 @@
...
 
 	/**
-	 * store settings for obj/package in DB
+	 * Save settings informations to DB  including field settings.
 	 * 
 	 * @param $settings Settings
 	 */
-	public function storePackageSettings(Settings $settings);
-
+	public function save(Settings $settings);
```
Comment (klees) [inline classes/Settings/DB.php]: What's the difference between save and update?

File: classes/Settings/Fields/DB.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+/* Copyright (c) 2016 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see ./LICENSE */
```
Comment (klees) [inline classes/Settings/Fields/DB.php]: No, you are not.

File: classes/Settings/Fields/DB.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace CaT\Plugins\Jill\Settings\Fields;
+
+
+/**
+ * Interface to handle field settings.
+ */
+interface DB {
```
Comment (klees) [inline classes/Settings/Fields/DB.php]: Bad name and bad description imho. This is the database for the package specific settings on objects, right?

File: classes/Settings/DB.php (hunk 1 above)
Comment (shecken) [inline classes/Settings/DB.php]: Save inserts a line into assignment table on creation of the repo object. Update makes an update on the table row, when you change the assigned package. It is not possible to assign two packages at the same time.

**Pull request merged**